File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/mapper/LibrarianMapper.java
package ru.elenacherev.librarymanager.mapper;

import ru.elenacherev.librarymanager.api.dto.Librarian;
import ru.elenacherev.librarymanager.domain.entity.LibrarianEntity;

public class LibrarianMapper {
    public static Librarian map(LibrarianEntity entity) {
        return Librarian.builder()
                .librarianId(entity.getLibrarianId())
                .build();
    }

    public static LibrarianEntity map(Librarian dto) {
        return map(dto, new LibrarianEntity());
    }

    public static LibrarianEntity map(Librarian dto, LibrarianEntity entity) {

        return entity;
    }
}

File: librarymanager-be/src/main/kotlin/ru/elenacherev/librarymanager/mapper/LibrarianMapper.kt
package ru.elenacherev.librarymanager.mapper

import ru.elenacherev.librarymanager.api.dto.Librarian
import ru.elenacherev.librarymanager.domain.entity.LibrarianEntity

fun LibrarianEntity.map() = Librarian(
    librarianId = librarianId
)

fun Librarian.map() = LibrarianEntity()

fun Librarian.map(entity: LibrarianEntity) = entity.also {
}